app/controllers/user_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class UserController < ApplicationController
+  def edit
+    update = @current_user.update(permitted_params)
+    head(:ok) if update
+  rescue ActiveRecord::RecordNotFound
+    head(:not_found) unless update
+  end
```

Comment:
Not_found error is already rescue in the application_controller. Could you delete this please

Reply:
Not_found error is rescued in authentication_controller, application_controller just raises an "unauthorized" status if User doesn't exist.

Should skip this application_controller's before_action and rewrite it inside user_controller?

Reply:
In the application controller we rescue all the ActiveRecord::NotFound whit this sentence `rescue_from ActiveRecord::RecordNotFound, with: -> { head(:not_found) } ` no matters in what action or controller this error is raised because we are inheriting from it.

And based on what your are saying, maybe we can delete from the authentication_controller the rescue for the NotFound Error. I'll check this and I'm going changing it if necessary